service/simpledb/unmarshall_error.go

```diff
@@ -0,0 +1,51 @@
+package simpledb
+
+import (
+	"encoding/xml"
+	"io"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+type xmlErrorDetail struct {
+	Code    string `xml:"Code"`
+	Message string `xml:"Message"`
+}
+
+type xmlErrorResponse struct {
+	XMLName   xml.Name         `xml:"Response"`
+	Errors    []xmlErrorDetail `xml:"Errors>Error"`
+	RequestID string           `xml:"RequestID"`
+}
+
+func unmarshalError(r *request.Request) {
```

Comment:
This looks similar to the S3's error handling. Was it the same?  We don't have to do it here, but it might be a good idea in the future to refactor this custom error handling to use the same code.